### PR TITLE
Interpolation Search in Dart

### LINKED ISSUE #5263: Interpolation Search in Dart
## 🚀 Feature

The Interpolation Search is an improvement over Binary Search for instances, where the values in a sorted array are uniformly distributed. Interpolation search may go to different locations according to the value of the key being searched. For example, if the value of the key is closer to the last element, interpolation search is likely to start search toward the end side.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by ritvij14**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

The interpolation search algorithm in Dart.
https://replit.com/@ritvij14/DSA-NeoAlgo#InterpolationSearch.dart

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #5263

### CODE REVIEW

File: Dart/search/InterpolationSearch.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+25
+32
+47
+56
+68
+Enter the number to be searched:
+47
+47 found at position: 4
+
+Time complexity: O(log(log n))
```
Comment (bhav09) [inline Dart/search/InterpolationSearch.dart:72]: Everything is fine, just mention Time complexity above Sample IO

**Review by bhav09 — CHANGES_REQUESTED** [review]
Minor Changes.

File: Dart/search/InterpolationSearch.dart (hunk 1 above)
Comment (ritvij14) [reply Dart/search/InterpolationSearch.dart:72]: @bhav09 done

**Comment by ritvij14** [comment]
merge conflicts resolved

File: Dart/search/InterpolationSearch.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+32
+47
+56
+68
+Enter the number to be searched:
+47
+47 found at position: 4
+
+Space complexity: O(1)
+ */
```
Comment (ankitaggarwal23) [inline Dart/search/InterpolationSearch.dart]: Rewrite it as,

```
/*

Sample Output:

Enter the size of list:
6
Enter the sorted list:
12
25
32
47
56
68
Enter the number to be searched:
47
47 found at position: 4

Time complexity: O(log(log n))
Space complexity: O(1)
*/

```

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Dart/search/InterpolationSearch.dart (hunk 2 above)
Comment (ritvij14) [reply Dart/search/InterpolationSearch.dart]: @ankitaggarwal23 done.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**